### DISCUSSION TITLE
Errors when update MediatR to version 11.0

### DISCUSSION THREAD

**Discussion by Hunglmc** [jasontaylordev/CleanArchitecture#797] [General]
Dear Sir.
Please help me how to fix errors  in ValidationBehaviour when i update  MediatR  to version 11.0

**Comment by hummer007** [comment]
Replace Handle method in ValidationBehavior with this 
```cs
    public async Task<TResponse> Handle(TRequest request, RequestHandlerDelegate<TResponse> next, CancellationToken cancellationToken)
    {
        if (!_validators.Any()) return await next();
        var context = new ValidationContext<TRequest>(request);

        var validationResults =
            await Task.WhenAll(_validators.Select(v => v.ValidateAsync(context, cancellationToken)));

        var failures = validationResults
            .SelectMany(r => r.Errors)
            .Where(f => f != null)
            .ToList();

        if (failures.Count != 0)
            throw new ValidationException(failures);

        return await next();
    }
```

**Reply by Hunglmc** [reply]
Thank you so much Sir .It woking.But i don't understanding problem.Can you tell me about different in this.I wish you have nice day.

**Reply by Hunglmc** [reply]
I think perhaps the order of the input values changed causing the error to occur

**Comment by neozhu** [comment]
refer to https://github.com/jbogard/MediatR/wiki/Migration-Guide-11.x-to-12.0
https://github.com/jbogard/MediatR/wiki/Migration-Guide-11.x-to-12.0
